src/backend/core/authentication/backends.py

```diff
@@ -104,3 +111,32 @@ def get_existing_user(self, sub, email):
                         _("Multiple user accounts share a common email.")
                     ) from e
         return None
+
+    @staticmethod
+    def compute_full_name(user_info):
+        """Compute user's full name based on OIDC fields in settings."""
+        full_name = " ".join(
+            filter(
+                None,
+                (
+                    user_info.get(field)
+                    for field in settings.OIDC_USERINFO_FULLNAME_FIELDS
+                ),
+            )
+        )
+        return full_name or None
+
+    @staticmethod
+    def update_user_if_needed(user, claims):
+        """Update user claims if they have changed."""
+
+        updated_claims = {
+            key: value
+            for key, value in claims.items()
+            if value is not None and value != getattr(user, key)
```

Comment:
Here's a clearer and more structured version of your comment:
There's an open question about filtering None values. Should we consider removing
an existing field as a valid update (e.g., when an identity provider explicitly
removes undesired data)? Current code could lead to data desynchronization.

Since we match users by email, different identity providers might return
conflicting data - one providing names while another returns blank values.
@sampaccoud @qbey  what's your take on this?

Reply:
I've seen this pattern in impress (I guess, and maybe in people) and I was wondering the same. 

I ended up with : we don't want to "delete" data, which as you say can lead to desynchronization but I guess we will see if it raises an issue in the future. 

Note: filtering out "None" does not mean filtering out Falsy (ie empty strings), should we? ^^

Reply:
I will update the code with `if not value`